File: GarageGo-BE/Models/ChiTietHDDV.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GarageGo_BE.Models
{
    [Table("CHITIET_HDDV")]
    public class ChiTietHDDV
    {
        [Key]
        public int MaCTDV { get; set; }

        [Required]
        public int MaHD { get; set; }

        [Required]
        public int MaDV { get; set; }

        public int SoLuong { get; set; } = 1;

        [Column(TypeName = "decimal(18,2)")]
        public decimal DonGia { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal ThanhTien { get; set; }

        [ForeignKey("MaHD")]
        public HoaDon? HoaDon { get; set; }

        [ForeignKey("MaDV")]
        public DichVu? DichVu { get; set; }
    }
}

File: GarageGo-BEJava/src/main/java/com/garagego/model/ChiTietHDDV.java
package com.garagego.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "CHITIET_HDDV")
@IdClass(ChiTietHDDVId.class)
public class ChiTietHDDV {

    @Id
    @Column(name = "MaHD")
    private Integer maHD;

    @Id
    @Column(name = "MaDV")
    private Integer maDV;

    @Column(name = "SoLuong")
    private Integer soLuong;

    @Column(name = "DonGia", precision = 18, scale = 2)
    private BigDecimal donGia;

    @Column(name = "ThanhTien", precision = 18, scale = 2)
    private BigDecimal thanhTien;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "MaHD", insertable = false, updatable = false)
    private HoaDon hoaDon;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "MaDV", insertable = false, updatable = false)
    private DichVu dichVu;
}
